src/createBtcrDid.cpp

```diff
@@ -0,0 +1,288 @@
+#include "satoshis.h"
+#include "classifyInputString.h"
+#include "libtxref/txref.h"
+#include "txid2txref.h"
+#include "t2tSupport.h"
+#include <bitcoinapi/bitcoinapi.h>
+#include <anyoption/anyoption.h>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace pt = boost::property_tree;
+
+
+struct TransactionData {
+    std::string inputString;
+    std::string outputAddress;
+    std::string network;
+    std::string wif;
+    std::string ddoRef;
+    double fee = 0.0;
+    unsigned int utxoIndex = 0;
+};
+
+
+std::string find_homedir() {
+    std::string ret;
+    char * home = getenv("HOME");
+    if(home != nullptr)
+        ret.append(home);
+    return ret;
+}
+
+int parseCommandLineArgs(int argc, char **argv, struct t2t::Config &config, struct TransactionData &transactionData) {
+
+    auto opt = new AnyOption();
+    opt->setFileDelimiterChar('=');
+
+    opt->addUsage( "" );
+    opt->addUsage( "Usage: createBtcrDid [options] <inputXXX> <changeAddress> <network> <WIF> <fee> <ddoRef>" );
+    opt->addUsage( "" );
+    opt->addUsage( " -h  --help                 Print this help " );
+    opt->addUsage( " --rpchost [rpchost or IP]  RPC host (default: 127.0.0.1) " );
+    opt->addUsage( " --rpcuser [user]           RPC user " );
+    opt->addUsage( " --rpcpassword [pass]       RPC password " );
+    opt->addUsage( " --rpcport [port]           RPC port (default: try both 8332 and 18332) " );
+    opt->addUsage( " --config [config_path]     Full pathname to bitcoin.conf (default: <homedir>/.bitcoin/bitcoin.conf) " );
+    opt->addUsage( " --utxoIndex [index]        Index # of which UTXO to use from the input transaction (default: 0) " );
+    opt->addUsage( "" );
+    opt->addUsage( "<inputXXX>      input: (bitcoin address, txid, txref, or txref-ext) needs at least slightly more unspent BTCs than your offered fee" );
+    opt->addUsage( "<outputAddress> output bitcoin address: will receive transaction change and be the basis for your DID" );
+    opt->addUsage( "<network>       the bitcoin network you want to use, 'main' or 'test'" );
+    opt->addUsage( "<WIF>           WIF representation of your private key" );
+    opt->addUsage( "<fee>           fee you are willing to pay (suggestion: >0.001 BTC)" );
+    opt->addUsage( "<ddoRef>        reference to a DDO you want as part of your DID (optional)" );
+
+    opt->setFlag("help", 'h');
+    opt->setCommandOption("rpchost");
+    opt->setOption("rpcuser");
+    opt->setOption("rpcpassword");
+    opt->setOption("rpcport");
+    opt->setCommandOption("config");
+    opt->setOption("utxoIndex");
+
+    // parse any command line arguments--this is to get the "config" option
+    opt->processCommandArgs( argc, argv );
+
+    // print usage if no options
+    if( ! opt->hasOptions()) {
+        opt->printUsage();
+        delete opt;
+        return -1;
+    }
+
+    // see if there is a bitcoin.conf file to parse. If not, continue.
```

Comment:
"testnet=1" is valid in config file.  This suggests it should be an optional argument.